src/WCSimPrimaryGeneratorAction.cc

```diff
@@ -414,6 +422,79 @@ void WCSimPrimaryGeneratorAction::GeneratePrimaries(G4Event* anEvent)
       SetBeamDir(dir);
       SetBeamPDG(pdg);
     }
+
+  else if (useNiBallEvt)
+    { //Pablo: add Ni Ball source
+        if ( !myNiBallGenerator ) {
+                myNiBallGenerator = new WCSimGenerator_NiBall(myDetector);
+                myNiBallGenerator->Initialize();
+                randGen = new TRandom3();
+        }
+        MyGPS->ClearAll();
+        fNiBallPosition[0] = niball_X;
+        fNiBallPosition[1] = niball_Y;
+        fNiBallPosition[2] = niball_Z;
+        for (int jj=0; jj<4; jj++) rn[jj] = randGen->Rndm();
```

Comment:
Yes, I can change that.

Reply:
does this relate to https://github.com/bquilain/WCSim/pull/13?

Reply:
I hadn't seen that. Personally I would recommend just avoiding using TRandom3 at all and use the Geant4 random generator throughout, to avoid e.g. accidentally re-seeding it anywhere else. If I had seen the other PR before it had been merged I would have recommended the same there. The only place it's used as far as I can find is [here](https://github.com/bquilain/WCSim/blob/hybridPMT/src/WCSimEventAction.cc#L282), which could be changed I think.

Reply:
Replaced.